pilopt/src/inliner.rs

```diff
@@ -1,10 +1,46 @@
+use itertools::Itertools;
 use powdr_constraint_solver::{
     constraint_system::ConstraintSystem, quadratic_symbolic_expression::QuadraticSymbolicExpression,
 };
 use powdr_number::FieldElement;
+use std::collections::{HashMap, HashSet};
 use std::fmt::Display;
 use std::hash::Hash;
 
+/// Keeps track of which constraints and bus interactions reference a given variable.
+/// Maps variables to constraint/bus interaction indices.
+struct VarIndex<V> {
```

Comment:
Can we use `IndexedConstraintSystem` from the constraint-solver crate here?

Reply:
done, much cleaner now!